From 0d4aa7aa5c7e756d079fd49a9d073c8099645de1 Mon Sep 17 00:00:00 2001
From: Xiyue Deng <manphiz@gmail.com>
Date: Mon, 30 Sep 2024 23:45:29 -0700
Subject: [PATCH] Add emacs-editing-modes metapackage for editing modes

* Recommends more well-known editing modes.
* Suggests lesser-known ones.
---
 debian/control | 52 ++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 52 insertions(+)

diff --git a/debian/control b/debian/control
index 1d3f97f..d471a4b 100644
--- a/debian/control
+++ b/debian/control
@@ -41,3 +41,55 @@ Description: Miscellaneous add-ons for Emacs
  .
  See /usr/share/doc/emacs-goodies-el/README.Debian.gz for a discussion
  of the disposition of previously bundled add-ons.
+
+Package: emacs-editing-modes
+Architecture: all
+Depends: ${misc:Depends}
+Recommends:
+ elpa-apache-mode,
+ elpa-bazel-mode,
+ elpa-bison-mode,
+ elpa-bpftrace-mode,
+ elpa-caml,
+ elpa-clojure-mode,
+ elpa-cmake-mode,
+ elpa-csv-mode,
+ elpa-dart-mode,
+ elpa-dockerfile-mode,
+ elpa-gnuplot-mode,
+ elpa-go-mode,
+ elpa-graphviz-dot-mode,
+ elpa-haskell-mode,
+ elpa-jinja2-mode,
+ elpa-js2-mode,
+ elpa-kotlin-mode,
+ elpa-lua-mode,
+ elpa-markdown-mode,
+ elpa-matlab-mode,
+ elpa-meson-mode,
+ elpa-nginx-mode,
+ elpa-php-mode,
+ elpa-protobuf-mode,
+ elpa-racket-mode,
+ elpa-rust-mode,
+ elpa-scala-mode,
+ elpa-sml-mode,
+ elpa-systemd,
+ elpa-vala-mode,
+ elpa-web-mode,
+ elpa-yaml-mode,
+Suggests:
+ elpa-ats2-mode,
+ elpa-elm-mode,
+ elpa-fountain-mode,
+ elpa-inform-mode,
+ elpa-kivy-mode,
+ elpa-pip-requirements,
+ elpa-puppet-mode,
+ elpa-qml-mode,
+ elpa-rainbow-mode,
+Description: Miscellaneous editing modes for Emacs
+ This is a metapackage that recommends or suggests various useful
+ Emacs editing modes including many programming modes, configuration
+ file modes, etc.  The more well-known modes are recommended, and the
+ lesser known ones are suggested.
-- 
2.45.2

